utils.py

```diff
@@ -1,43 +1,64 @@
 #!/usr/bin/env python
 
 import sys
+import array
 import pygame
 
 import wx
 wx.App()
 
 import numpy as np
 
+from PIL import Image
+
 from skimage.color import rgb2gray
 from skimage.transform import resize
 from skimage.io import imread
+from skimage.util import img_as_float
 
 import matplotlib.pyplot as plt
 import matplotlib.image as mpimg
 
 
-IMG_W = 200
-IMG_H = 66
-
 
 def take_screenshot():
     screen = wx.ScreenDC()
-    size = screen.GetSize()
-    bmp = wx.Bitmap(size[0], size[1])
+    bmp = wx.Bitmap(ImageHelper.SRC_W, ImageHelper.SRC_H)
     mem = wx.MemoryDC(bmp)
-    mem.Blit(0, 0, size[0], size[1], screen, 0, 0)
-    return bmp.GetSubBitmap(wx.Rect([0,0],[615,480]))
+    mem.Blit(0, 0, ImageHelper.SRC_W, ImageHelper.SRC_H, screen, ImageHelper.OFFSET_X, ImageHelper.OFFSET_Y)
+    return bmp
+
 
 
 def prepare_image(img):
     if(type(img) == wx._core.Bitmap):
-        buf = img.ConvertToImage().GetData()
-        img = np.frombuffer(buf, dtype='uint8')
+        img.CopyToBuffer(arr)
+        img = np.frombuffer(arr, dtype=np.uint8)
+
+    img = img.reshape(ImageHelper.SRC_H, ImageHelper.SRC_W, ImageHelper.SRC_D)
+
+    im = Image.fromarray(img)
+    im = im.resize((ImageHelper.IMG_W, ImageHelper.IMG_H))
+
+    im_arr = np.frombuffer(im.tobytes(), dtype=np.uint8)
+    im_arr = im_arr.reshape((ImageHelper.IMG_H, ImageHelper.IMG_W, ImageHelper.IMG_D))
+
+    return img_as_float(im_arr)
+
+
+class ImageHelper:
+    SRC_W = 615
+    SRC_H = 480
+    SRC_D = 3
+
+    OFFSET_X = 0
+    OFFSET_Y = 0
 
-    img = img.reshape(480, 615, 3)
-    img = resize(img, [IMG_H, IMG_W])
+    IMG_W = 200
+    IMG_H = 66
+    IMG_D = 3
 
-    return img
+    image_array = array.array('B', [0] * (SRC_W * SRC_H * SRC_D));
```

Comment:
Yeah that is a great idea! I was originally thinking that the screenshot class was going to own the prepare_image function but after looking at the way you did it I much prefer your way :)